invoicing/services.py

```diff
@@ -12,31 +15,50 @@
 logger = getLogger(__name__)
 
 
+TALPA_EMAIL_SUBJECT_TEMPLATE = "{sender_id} aineisto"
+TALPA_EMAIL_CONTENT_TEMPLATE = (
+    "{sender_id} myyntireskontra-aineistoa lähetetty {count} kpl {date_time}"
+)
+
+
 def send_needed_installments_to_sap() -> int:
     installments = ApartmentInstallment.objects.sending_to_sap_needed()
     logger.debug(f"Installment IDs: {[i.pk for i in installments]}")
     num_of_installments = installments.count()
+    ts = timezone.now()
     if num_of_installments:
         xml = generate_installments_xml(installments)
         logger.debug(f"Generated XML:\n{xml.decode('utf-8')}")
-        send_xml_to_sap(xml)
+        send_xml_to_sap(xml, ts)
         installments.set_sent_to_sap_at()
         for installment in installments:
             audit_logging.log(None, Operation.UPDATE, installment)
-    return num_of_installments
+    return num_of_installments, ts
 
 
-def generate_sap_xml_filename() -> str:
-    return (
-        settings.SAP_SFTP_FILENAME_PREFIX
-        + f"{timezone.now().strftime('%Y%m%d%H%M%S')}"
-        + ".xml"
+def send_email_notification_to_talpa(count: int, ts: datetime):
+    email = EmailMessage(
+        subject=TALPA_EMAIL_SUBJECT_TEMPLATE.format(sender_id=SAP["SENDER_ID"]),
+        body=TALPA_EMAIL_CONTENT_TEMPLATE.format(
+            sender_id=SAP["SENDER_ID"],
+            count=count,
+            date_time=ts.strftime("%Y%m%d%H%M%S"),
+        ),
+        to=[TALPA_EMAIL],
+        reply_to=[TALPA_EMAIL_REPY_TO],
     )
+    return email.send()
+
+
+def generate_sap_xml_filename(ts: datetime) -> str:
+    return settings.SAP_SFTP_FILENAME_PREFIX + f"{ts.strftime('%Y%m%d%H%M%S')}" + ".xml"
 
 
-def send_xml_to_sap(xml: bytes, filename: str = None) -> None:
+def send_xml_to_sap(
+    xml: bytes, filename: str = None, ts: datetime = timezone.now()
```

Comment:
Here `timezone.now()` will be evaluated only once, so the default value will be the same for every call of `send_xml_to_sap()`, which is probably not what we want 🙂